lib/web_ui/lib/src/engine/text/paragraph.dart

```diff
@@ -24,70 +24,7 @@ class EngineLineMetrics implements ui.LineMetrics {
     required this.left,
     required this.baseline,
     required this.lineNumber,
-  })  : displayText = null,
-        ellipsis = null,
-        startIndex = -1,
-        endIndex = -1,
-        endIndexWithoutNewlines = -1,
-        widthWithTrailingSpaces = width,
-        boxes = <RangeBox>[],
-        spaceBoxCount = 0,
-        trailingSpaceBoxCount = 0;
-
-  EngineLineMetrics.rich(
```

Comment:
Note for self: Instead of adding a bunch of properties to EngineLineMetrics so the `rich` constructor can be used, this PR converts `.rich` in its own class, that extends EngineLineMetrics.